### PR TITLE
Add dedup_clone

### PR DESCRIPTION
**Pull request by p-salazarhamm**
Hey Kev,
Adding annotated dedup_clone.R script with test.

### CODE REVIEW

File: tublerone/R/dedup_clones.R [hunk 1]
```diff
@@ -1,18 +1,39 @@
...
+#' @export 
 
-identicals = list()
-for (row in rownames(snp_pairwise)) {
-  identicals[[row]] = sort(colnames(snp_pairwise)[which(snp_pairwise[row, ] == 0)])
+dedup_clones = function(snp_pairwise, minimum_dist = 0) {
+
+  identicals = list()
+  for (row in rownames(snp_pairwise)) {
+    identicals[[row]] = sort(colnames(snp_pairwise)[which(snp_pairwise[row, ] == minimum_dist)])
```
Comment (amsesk) [inline tublerone/R/dedup_clones.R]: The logic here is wrong. As written, duplicates will only be removed if the pairwise distance is equal to `minimum_dist`. This is not the intended functionality.

**Comment by amsesk** [comment]
Thanks for the PR. The function looks good except that the conditional expression on L26 is wrong. See review comment above. It looks like it was not updated with the addition of the `minimum_dist` parameter with 2005e419bf67957cf466c05881c095d23c497b79.

We could avoid these kinds of mistakes by implementing unit tests for functions in this library, which is generally a good idea that I have neglected thus far.

https://cran.r-project.org/web/packages/box/vignettes/testing.html

In any case, I can merge once the logic for `minimum_dist` is fixed.

**Comment by p-salazarhamm** [comment]
I have corrected == minimum_dist to <= minimum_dist so that all distances below a minimum will be removed.

**Comment by amsesk** [comment]
Looks good. Let's add official tests to this and other functions when we get some time to spring clean.

**Pull request merged**